### PR TITLE
added warning modals to unsaved bounty and profile pages

### PR DESCRIPTION
**Pull request by deeayeen**

### CODE REVIEW

**Comment by corwinharrell** [comment]
@deeayeen thanks for submitting this! We will take some time to review and will leave any feedback that might need addressing here before we accept your submission and approve your payout.

@mbeylin If you have some time and don't mind, I'd love a second pair of eye's in reviewing this and doing some QA. Let me know if you have any thoughts around anything in particular.

**Comment by corwinharrell** [comment]
@deeayeen We have another Bounties Network community member who also applied to this bounty who has asked if it would be ok to contribute to reviewing the pull-request (just as a learning opportunity) as he is currently sharpening his react skillset. Do you mind the extra set of eyes?

**Comment by deeayeen** [comment]
Sure, let me know if anything needs any explanation, the react-router can
get confusing.

**Comment by corwinharrell** [comment]
@deeayeen awesome, thanks for being flexible!

@adnfx2 feel free to jump in 😄

**Comment by adnfx2** [comment]
Thanks guys for being so nice and letting me participate here. 
Here I go:

1) In CreateBounty you defined the following states: dirty, isit and showModal, I see the point of creating dirty, and it's also being used down there. But isit and showModal are not being used anywhere in the code, same thing happened with the following methods defined in this class: showModal, hideModal and hideModalTemp. So my question is why are they defined in first place? Are they really needed?

2) I tested the code a bit and I found the following behaviors:
- In CreateBounty form, when clicking on: - create bounty or create draft or update draft - to submit the form, the modal is displayed, I'm guessing this should not happen(Please correct if I'm wrong).
- In Settings when editing EmailPreferences, no modal is displayed if I attempt to leave, only when I modified UserSettings the modal will be potentially displayed. Also in UserSettings if I make an edition and successfully gets saved after clicking update profile, when attempting to leave that page, a modal will be displayed, this prolly is not the intended behavior.

Hope that my observations can become useful to you guys, or if you think I might have done something wrong when testing, please let me know.

**Comment by deeayeen** [comment]
Perfect, that is very useful feedback Adrian, thank you.

Corwin:
To sum up
1) I declared some variables during testing that I need to delete.
2) I am currently blocking all routing if the form has unsaved changes,
including the submit form route. I need to whitelist certain routes from
being blocked.
3) Only added modal to user settings, need to add modal when editing email
preferences as well.

I will submit these fixes shortly.

**Comment by adnfx2** [comment]
Hey man that's awesome!, one more thing, the content of the modal like it's title: "Discard Changes?" it's hard coded, I think you should use intl.get to place the content there from us_EN.json

**Comment by deeayeen** [comment]
Submitted updated PR.

**Comment by mbeylin** [comment]
I have some notes on this PR:
- When editing an existing bounty, I can click the "increase payout" link to be taken to the bounty detail page. Instead of seeing our custom modal, I'm presented with this one instead:
Screen Shot 2019-07-02 at 1 11 24 PM
This should be replaced with our custom modal.
- Some of the if statements you've added are quite messy. For instance:
```
        if (
	      this.state.dirty &&
	      !isEditing &&
	      (this.state.dirty && !isDraft) &&
	      (this.state.dirty && this.state.submitNotPressed)
	    )
``` 

should be simplified to:
```if (this.state.dirty &&  !isEditing && !isDraft && this.state.submitNotPressed)```
Please simplify this and other if statements in the codebase to maintain simplicity and efficiency.
- There's still a variable `handleUser` from UserSettings.js that is assigned a value and never used.

**Comment by mbeylin** [comment]
To test out the editing of bounties, you can use the command `yarn start-rinkeby` and run a version of the explorer that points to the Rinkeby test network, on which you can easily create a new bounty without having to pay for gas on the main net.

File: src/containers/CreateBounty/index.js [hunk 1]
```diff
@@ -64,23 +71,145 @@ class CreateBountyComponent extends React.Component {
...
+              return (
+                <Modal
+                  dismissable
+                  size={'medium'}
+                  fixed
+                  visible={true}
+                  onClose={onCancel}
+                >
+                  <Modal.Header closable />
+                  <Modal.Body>
```
Comment (corwinharrell) [inline src/containers/CreateBounty/index.js]: For our modal component, formatting is built in, and you shouldn't need to add any additional classNames or styles. For reference, take a look at `src/containers/Bounty/components/modals/ContributeFormModal.js` to see how `Modal.Message` and `Modal.Description` are used to determine how different elements of the modal are structured and styled. Please remove any inline styles and HTML tags and defer to composing the modal in this way.

As others has also mentioned, please replace the text content with copy added to `src/i18n/en_US.json`.

File: src/containers/CreateBounty/index.js [hunk 2]
```diff
@@ -64,23 +71,145 @@ class CreateBountyComponent extends React.Component {
...
```
Comment (corwinharrell) [inline src/containers/CreateBounty/index.js]: For modal formatting, see comments above.

**Comment by deeayeen** [comment]
Submitted a PR with the following changes:

- Removed unused variables
- Improved if statements
- Corrected modal formatting
- Internationalized modal text

When creating a new bounty on the Rinkeby network it loads infinitely.

**Comment by deeayeen** [comment]
It seems as though the “increase funds” is an external redirect,
react-router has no way to catch these kinds of redirects (refreshing the
page, entering a new url, etc) so I integrated the pop up prompt you
mentioned for these special cases. I hope this is alright. Other than that
I have addressed all issues you guys have mentioned, please let me know
what else I can help with.

**Comment by corwinharrell** [comment]
@mbeylin any thoughts on the above?

**Pull request merged**